src/app/frontend/common/autorefresh/autorefresh_directive.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {AutoRefreshController} from './autorefresh_controller';
+import {DEFAULT_DELAY} from './autorefresh_controller';
+/**
+ * Provides upload file mechanism and updates the model with file parameter for the selected file.
+ * @param {!./poll_service.PollService} kdPoll
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function autoRefreshDirective(kdPoll) {
+  return {
+    scope: {},
+    bindToController: {
+      // Resource object to query data.
+      'source': '=',
+      // The the model object which is updated in each poll.
+      'target': '=',
+      // Type of the page like detail list, list-without-namespace, detail page and etc.
+      'type': '@?',
+      // Polling timeout in milliseconds.
+      'delay': '@?',
+    },
+    controller: AutoRefreshController,
+    controllerAs: 'ctrl',
+    templateUrl: '',
+    link: function(scope, element, attrs) {
+      /**
+       * @type {./poll_service.PollService}
+       * @private
+       */
+      let kdPoll_ = kdPoll;
+
+      // Cancel all polls in case of state change to avoid accumulated polls.
+      scope.$on('$stateChangeStart', () => {
+        kdPoll_.removeAll();
```

Comment:
So this is the place where you fixed the failure scenario, right?
Indeed, now it works for me :)
Still, I have a feeling that the code would be a little safer if you just passed the "options" thing from poll() to start() and stop() as a parameter, rather than introducing "currentOptions" as a de-facto global variable. (I'm fine about "optionsList"). Or am I missing something? Anyway, I won't insist on that.